packages/plexus/README.md

```diff
@@ -1,13 +1,914 @@
+
+Generally, the value of these can be either an object of props to set on the target element, a function which will generate either `null` or an object of props to set on the target, or an array either of these.
+
+```tsx
+const graphClassName = { className: 'LeGraphOlogy' };
+// ignoring the parameters that are passed to the factory function, for now...
+const generatePaddingStyle = () => {
+  style: {
+    padding: `${(Math.random() * 10).toFixed()}px`;
+  }
+};
+
+// All three of these are valid:
+
+// Set only the CSS class
+const ok = (
+  <LayeredDigraph
+    edges={edges}
+    vertices={vertices}
+    setOnGraph={graphClassName}
+    // etc...
+  />
+);
+
+// Set only the random padding
+const alsoOk = (
+  <LayeredDigraph
+    edges={edges}
+    vertices={vertices}
+    setOnGraph={generatePaddingStyle}
+    // etc...
+  />
+);
+
+// Set both the CSS class and the random padding
+const allOfTheAbove = (
+  <LayeredDigraph
+    edges={edges}
+    vertices={vertices}
+    setOnGraph={[graphClassName, generatePaddingStyle]}
+    // etc...
+  />
+);
+```
+
+## API
+
+- [Common types](#common-types)
+  - [TExtension](#textension)
+  - [TVertexKey](#tvertexkey)
+  - [TVertex](#tvertex)
+  - [TEdge](#tedge)
+  - [TLayoutVertex](#tlayoutvertex)
+  - [TLayoutEdge](#tlayoutedge)
+  - [TLayoutGraph](#tlayoutgraph)
+  - [ELayoutPhase](#elayoutphase)
+  - [TRendererUtils](#trendererutils)
+  - [TExposedGraphState](#texposedgraphstate)
+  - [TContainerPropsSetter](#tcontainerpropssetter)
+  - [TDefEntry](#tdefentry)
+- [`LayoutManager` options](#layoutmanager-options)
+- [`LayeredDigraph` props](#layereddigraph-props)
+- [Layer configuration objects](#layer-configuration-objects)
+  - [Common to all layers](#common-to-all-layers)
+  - [Common to SVG layers](#common-to-svg-layers)
+  - [Measurable nodes layer](#measurable-nodes-layer)
+  - [Nodes layer](#nodes-layer)
+  - [Edges layer](#edges-layer)
+  - [HTML and SVG layers group](#html-and-svg-layers-group)
+- [Builtin props factories](#builtin-props-factories)
+  - [classNameIsSmall](#classnameissmall)
+  - [scaleProperty.opacity](#scalepropertyopacity)
+  - [scaleProperty.strokeOpacity](#scalepropertystrokeopacity)
+  - [scaleProperty](#scaleproperty)
+
+### Common types
+
+#### `TExtension`
+
+The default type parameter for extending `TVertex` and `TEdge`.
+
+```tsx
+type TExtension = Record<string, unknown>;
+```
+
+#### `TVertexKey`
+
+The type for the `key` field on vertices.
+
+```tsx
+type TVertexKey = string | number;
+```
+
+#### `TVertex`
+
+The type for the data underlying vertices.
+
+```tsx
+type TVertex<T = TExtension> = T & {
+  key: TVertexKey;
+  // TODO(joe): remove label
+  label?: React.ReactNode;
+};
+```
+
+#### `TEdge`
+
+The data that underlies edges in a graph.
+
+```tsx
+type TEdge<T = TExtension> = {
+  from: TVertexKey;
+  to: TVertexKey;
+  isBidirectional?: boolean;
+  label?: React.ReactNode;
+  data?: T;
+};
+```
+
+#### `TLayoutVertex`
```

Comment:
If this, `TLayoutEdge`, `TLayoutGraph`, and `ELayoutPhase` are all only available through `TExposedGraphState`, should they be displayed similarly to `setOnNode`, `renderNode`, and `measureNode`? i.e. instead of being their own headers, they can exist under `TExposedGraphState`

When I first read these types I wondered if/where they were exposed, so the context would probably help.

Reply:
Thanks, will check it out.

Reply:
I shifted these around a bit. Thanks for the observation.

Reply:
the reworked categories are clearer 👍